### PR TITLE
Fixed tests for DomainCountLimiter

### PR DESCRIPTION
**Pull request by priettt**
## Goal

Moved the tests that were present in EmbraceNetworkLoggingService to DomainCountLimiter

### CODE REVIEW

**Comment by priettt** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/757?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#757** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/757?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#755** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/755?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#749** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/749?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#748** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/748?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#747** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/747?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#746** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/746?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#745** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/745?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @priettt and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/EmbraceDomainCountLimiter.kt [hunk 1]
```diff
@@ -43,15 +43,15 @@ internal class EmbraceDomainCountLimiter(
...
-            if (countPerSuffix.requestCount < limit) {
-                return true
-            }
-
             // Track the number of calls for each domain (or configured suffix)
             suffix?.let {
                 callsPerDomainSuffix[it] = NetworkSessionV2.DomainCount(countPerSuffix.requestCount + 1, limit)
             }
+
+            // Exclude if the network call exceeds the limit
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/EmbraceDomainCountLimiter.kt:51]: why the change of order? did it not matter before or something has changed?

**Review by bidetofevil — APPROVED** [review]
LGTM

**Comment by bidetofevil** [comment]
### Merge activity

* **Apr 18, 7:50 PM EDT**: @bidetofevil started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/757).
